mantidimaging/gui/windows/main/test/view_test.py

```diff
@@ -329,18 +333,51 @@ def test_show_nexus_load_dialog_calls_show(self):
         nexus_load_dialog.assert_called_once()
         nexus_load_dialog.return_value.show.assert_called_once()
 
-    @mock.patch("mantidimaging.gui.windows.main.view.CommandLinePath")
+    @mock.patch("mantidimaging.gui.windows.main.view.CommandLineArguments")
     @mock.patch("mantidimaging.gui.windows.main.view.WelcomeScreenPresenter")
     @mock.patch("mantidimaging.gui.windows.main.view.MainWindowPresenter")
-    def test_load_path_from_command_line(self, main_window_presenter, welcome_screen_presenter, command_line_path):
-        command_line_path.return_value.path.return_value = test_path = "./"
+    def test_load_path_from_command_line(self, main_window_presenter, welcome_screen_presenter, command_line_args):
+        command_line_args.return_value.path.return_value = test_path = "./"
         MainWindowView()
         main_window_presenter.return_value.load_stacks_from_folder.assert_called_once_with(test_path)
 
-    @mock.patch("mantidimaging.gui.windows.main.view.CommandLinePath")
+    @mock.patch("mantidimaging.gui.windows.main.view.CommandLineArguments")
     @mock.patch("mantidimaging.gui.windows.main.view.WelcomeScreenPresenter")
     @mock.patch("mantidimaging.gui.windows.main.view.MainWindowPresenter")
-    def test_no_path_argument_set(self, main_window_presenter, welcome_screen_presenter, command_line_path):
-        command_line_path.return_value.path.return_value = ""
+    def test_command_line_no_path_argument_set(self, main_window_presenter, welcome_screen_presenter,
+                                               command_line_args):
+        command_line_args.return_value.path.return_value = ""
         MainWindowView()
         main_window_presenter.return_value.load_stacks_from_folder.assert_not_called()
+
+    @mock.patch("mantidimaging.gui.windows.main.view.CommandLineArguments")
+    @mock.patch("mantidimaging.gui.windows.main.view.WelcomeScreenPresenter")
+    @mock.patch("mantidimaging.gui.windows.main.view.MainWindowPresenter")
+    def test_command_line_show_filters_window(self, main_window_presenter, welcome_screen_presenter, command_line_args):
+        command_line_args.return_value.operation.return_value = command_line_filter = "Median"
+        MainWindowView()
+        main_window_presenter.return_value.show_operation.assert_called_once_with(command_line_filter)
```

Comment:
This test and a couple of the other new ones show the reconstruction window. You might need to add a sleep at the end to see it.

Looks like its because `MainWindowView.__init__()` `CommandLineArguments` is a mock, so `if args.recon()` evaluates true.

I think you'll need to set the mock return values for all three options in each of those tests.